soroban-env-host/src/host.rs

```diff
@@ -1002,6 +1002,7 @@ impl Host {
                 let max_balance = i64::MAX;
                 (min_balance, max_balance)
             };
+            let max_balance = min(max_balance, tl.limit);
```

Comment:
This excludes the buying liabilities from the limit, rather than reduces the buying liabilities from the limit. Is that correct?

Reply:
It seems suspicious to me because this would, I think, mean that you could exceed your limit by canceling things that create buying liabilities.

Reply:
Yeah you're right. `max_balance` should be initialized with `tl.limit` instead of `i64::MAX`.

Reply:
I think you are right, changed this to apply limit before liabilities.